### PR TITLE
Fix isDefinedAt bug in concatOrNotFound

### PR DESCRIPTION
**Pull request by raboof**
Refs #886

### CODE REVIEW

**Comment by jrudolph** [comment]
Without digging in more deeply, it's hard to understand what was wrong and how you fixed it. Could you give a short explanation?

File: interop-tests/src/test/scala/akka/grpc/scaladsl/CountingGreeterServiceImpl.scala [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+import java.util.concurrent.atomic.AtomicInteger
+
+import scala.concurrent.Future
+
+import akka.NotUsed
+import akka.stream.scaladsl.Source
+
+import example.myapp.helloworld.grpc.helloworld._
+
+class CountingGreeterServiceImpl extends GreeterService {
```
Comment (jrudolph) [inline interop-tests/src/test/scala/akka/grpc/scaladsl/CountingGreeterServiceImpl.scala]: Is this one covering the bug?

File: runtime/src/main/scala/akka/grpc/scaladsl/ServiceHandler.scala [hunk 2]
```diff
@@ -29,31 +27,13 @@ object ServiceHandler {
...
-    handler(handlers: _*)
-
-  /**
-   * Creates a `HttpRequest` to `HttpResponse` handler for gRPC services that can be used in
-   * for example `Http().bindAndHandleAsync` for the generated partial function handlers:
-   *  - The generated handler supports the `application/grpc` media type.
-   *  - If the request is for a handled service, but with an invalid media type, then a _415: Unsupported Media Type_ response is produced.
-   *  - Otherise if the request is not handled by one of the provided handlers, a _404: Not Found_ response is produced.
-   */
-  def handler(handlers: PartialFunction[HttpRequest, Future[HttpResponse]]*): HttpRequest => Future[HttpResponse] = {
```
Comment (jrudolph) [inline runtime/src/main/scala/akka/grpc/scaladsl/ServiceHandler.scala]: Is that one only internal so that it can just be removed?

**Comment by raboof** [comment]
The previous implementation used `servicesHandler.isDefinedAt`, but depending on the implementation of the `PartialFunction` (including in this case), `isDefinedAt` works by actually calling the function and seeing if it returns...

File: interop-tests/src/test/scala/akka/grpc/scaladsl/CountingGreeterServiceImpl.scala (hunk 1 above)
Comment (raboof) [reply interop-tests/src/test/scala/akka/grpc/scaladsl/CountingGreeterServiceImpl.scala]: (no, sorry, that was a cleanup I made while reproducing the bug in a testcase, but I ended up testing it with another project..)

File: runtime/src/main/scala/akka/grpc/scaladsl/ServiceHandler.scala (hunk 2 above)
Comment (ennru) [reply runtime/src/main/scala/akka/grpc/scaladsl/ServiceHandler.scala]: No, this it public API in 0.8

**Pull request merged**